### DISCUSSION TITLE
New AntiXray mod (engine 4)

### DISCUSSION THREAD

**Discussion by groundbreakingmc** [PaperMC/Paper#10660] [Ideas]
### Is your feature request related to a problem?

My problem is caused by the fact that the blocks are replaced either with stone or with already prepared ones, and I want that the replacement will  look natural

### Describe the solution you'd like.

I want the blocks that need to be hidden to be replaced by those blocks that are on the bottom or if there is empty, then with the block that is on the side or on top

### Describe alternatives you've considered.

I tried using Orebfuscator and decided to switch to RayCastAntiXray and it doesn't have such a feature as I wont

### Other

For exmple this is without antixray
![photo_5310101465613853004_y](https://github.com/PaperMC/Paper/assets/65243837/5b29c773-493e-4087-94ec-c566bb0cf8a1)
![photo_5310101465613853007_y](https://github.com/PaperMC/Paper/assets/65243837/8d68aca0-a982-4054-b7e3-05d3971ad2be)

This is paper angine mod 1
![photo_5310101465613853005_y](https://github.com/PaperMC/Paper/assets/65243837/12f0607a-da2a-41d6-ad84-8f50f83b8252)
![photo_5310101465613853008_y](https://github.com/PaperMC/Paper/assets/65243837/10eaca7b-a199-4e71-8f5b-0992803d83d8)

This is my engine mod 4
![photo_5310101465613853006_y](https://github.com/PaperMC/Paper/assets/65243837/aa1d989a-60c6-4fb2-9a3e-acc10eb90c77)
![photo_5310101465613853009_y](https://github.com/PaperMC/Paper/assets/65243837/332ce8c1-2aa8-428e-ab88-f05569fe8a25)

**Comment by notTamion** [comment]
i feel like this is hard to properly implement. what if the block is of the same type? that would basically render the anti xray useless for certain scenarios since the ore would still be shown. and since basically no xray shows stone/netherrack blocks anyways you could just use engine mode 1

**Reply by groundbreakingmc** [reply]
engine mode 1 I can’t use this mod because all hidden blocks are "visible" if, for example, there is andesite or another block around that is not stone, but what if I need to hide blocks in the end world or nether?
